src/connection_string.ts

```diff
@@ -394,6 +394,16 @@ export function parseOptions(
     }
 
     mongoOptions.credentials.validate();
+
+    // Check if the only auth related option provided was authSource, if so we can remove credentials
```

Comment:
The PR description and the PR title both say that only test additions are included; is this supposed to be a `fix:`? If so, can we include more context in the PR desciption?

Reply:
done!